src/librustc_trans/debuginfo/mod.rs

```diff
@@ -274,6 +274,16 @@ pub fn create_function_debug_context<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
     let function_name = CString::new(name).unwrap();
     let linkage_name = CString::new(linkage_name).unwrap();
 
+    let mut flags = FlagPrototyped as c_uint;
+    match *cx.sess().entry_fn.borrow() {
+        Some((id, _)) => {
+            if instance == Instance::mono(cx.shared(), cx.tcx().map.local_def_id(id)) {
```

Comment:
Oh, that makes sense. Though you could probably make it shorter be wrapping `id` in an `Option` and then comparing `Option` against `Option`. Or do something like this:
```rust
let entry_fn_id = *cx.sess().entry_fn.borrow();
if entry_fn_id.is_some() && local_id == entry_fn_id {
    flags |= ...;
}
```